compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -641,6 +628,15 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
 
   /** Display spaces */
   def show(s: Space): String = {
+    def params(tp: Type): List[Type] = {
```

Comment:
This function seems to be mostly doing unnecessary work - the actual types returned are never used, and the only thing that matters is the length of the returned list.